File: EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/ClassroomResourceAllocation.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class ClassroomResourceAllocation : BaseAuditableEntity
{
    public long ClassroomId { get; set; }
    public string ResourceNameAr { get; set; } = string.Empty;
    public string ResourceCode { get; set; } = string.Empty;
    public int ResourceType { get; set; } // 1=Projector, 2=SmartBoard, 3=AirConditioner, 4=DeskSet
    public int Quantity { get; set; }
    public DateTime AssignedDate { get; set; } = DateTime.UtcNow;
    public string? ConditionStatus { get; set; }
    public string? ResourceNameEn { get; set; }
    public string? AssetSerialNumber { get; set; }
    public decimal UnitPurchaseCost { get; set; }
    public DateTime? LastInspectionDate { get; set; }
    public DateTime? NextMaintenanceDate { get; set; }

    // Navigation Property
    public virtual Classroom? Classroom { get; set; }
}

File: EduMS-Frontend/edums-frontend-DDD-System/src/app/modules/school-admin/interfaces/classroom-resource-allocation.ts
import type { ResourceType } from './common';

export interface ClassroomResourceAllocation {
  id: number;
  classroomId: number;
  resourceNameAr: string;
  resourceCode: string;
  resourceType: ResourceType;
  quantity: number;
  assignedDate: string;
  conditionStatus: string | null;
  resourceNameEn: string | null;
  assetSerialNumber: string | null;
  unitPurchaseCost: number;
  lastInspectionDate: string | null;
  nextMaintenanceDate: string | null;
}

export type CreateClassroomResourceAllocationDto = Omit<ClassroomResourceAllocation, 'id' | 'conditionStatus'>;

export type UpdateClassroomResourceAllocationDto = Omit<ClassroomResourceAllocation, 'conditionStatus'>;
